### ISSUE TITLE
http: no Host in request URL

### ISSUE THREAD

**Issue by mobil-app-explorer** [stripe/stripe-cli#733]
In dashboard, the webhook url is set as: 
https://example.com/stripe-payment/webhook/checkout-session-completed

I used this command to listen: 
stripe listen --load-from-webhooks-api --forward-to localhost:5000

and got the following error message:
 [ERROR] Failed to POST: Post "http:///stripe-payment/webhook/checkout-session-completed": http: no Host in request URL

Your help is appreciated to get me around this issue, thanks in advance!

**Comment by tomer-stripe** [comment]
Hey! So the `--load-from-webhooks-api` will load your dashboard webhooks configuration and forward to that as well. example.com won't have that actual handler setup so the CLI won't be able to send data there.

**Comment by pepin-stripe** [comment]
@mobil-app-explorer can you give a bit more context on your OS, stripe CLI version and terminal you are using? 

I've created a webhook endpoint with the same URL as yours and ran the same command you shared and it seems to work on macOS with the latest CLI version (1.7.0) so I'm trying to understand what may be happening on your side.

The error you are receiving implies that the `--forward-to` argument received an empty string instead of the `localhost:5000` value passed, can you try again and make sure that nothing is truncating your command? Maybe try typing it manually, or change the order of args (`stripe listen --forward-to localhost:5000 --load-from-webhooks-api`) and see if the same issue persists.

**Comment by mobil-app-explorer** [comment]
@pepin-stripe Thanks for looking into this issue, I manually typed "stripe listen --forward-to localhost:5000 --load-from-webhooks-api", still got the same error message. Here are my system info:
Stripe:  version 1.7.0
OS: 
     Edition	Windows 10 Home
     Version	21H1
     OS build	19043.1165
Terminals: tried from both windows command line and power shell, same error

Right now, I am using other alternative solution, which is not to use "load-from-webhooks-api", and it is working well, so this is not urgent at all. Let me know if any other info needed, thanks again!

**Comment by ties-s** [comment]
I have the same problem. Tried all arg orders.
`[ERROR] Failed to POST: Post "http:///stripe-connect-webhook": http: no Host in request URL`

```
macOS 11.4
stripe 1.7.0 (homebrew)
```

Tried:
```
stripe listen --forward-to=local.example.com --load-from-webhooks-api
stripe listen --load-from-webhooks-api --forward-to=local.example.com
stripe listen -a -f local.example.com
stripe listen -f -a local.example.com
```

**Comment by pepin-stripe** [comment]
Thanks for the additional info @mobil-app-explorer and @ties-s . We're not able to repro on our side so trying to dive a bit deeper to figure out what the problem is.

Can you please provide an example event id (`evt_xxxxx`) that you would have wanted to received on one of those webhook endpoint?

Thanks!

**Comment by pepin-stripe** [comment]
@mobil-app-explorer  and @ties-s, I think the issue was found and fixed by a user in the PR #727.

Can you both confirm that the issue is for a _connect_ webhook endpoint?

**Comment by mobil-app-explorer** [comment]
Thanks for looking into this! Yes, when I saw this issue, it was connect webhook endpoint.

**Comment by pepin-stripe** [comment]
Great to hear! I'll update this once the PR is merged and we release a new patch on the CLI.

**Comment by pepin-stripe** [comment]
@mobil-app-explorer we just released `v1.7.2` which should fix your issue once you're upgrade your local install. 

I'm going to close this, but please feel free to re-open if the issue persists!
